ballerina/SQLClient.bal

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/sql;
+
+public client class SQLClient {
+
+    private final sql:Client dbClient;
+
+    private string entityName;
+    private sql:ParameterizedQuery tableName;
+    private map<FieldMetadata> fieldMetadata;
+    private string[] keyFields;
+
+    public function init(string entityName, sql:ParameterizedQuery tableName, map<FieldMetadata> fieldMetadata, string[] keyFields, sql:Client dbClient) returns error? {
+        self.entityName = entityName;
+        self.tableName = tableName;
+        self.fieldMetadata = fieldMetadata;
+        self.keyFields = keyFields;
+        self.dbClient = dbClient;
+    }
+
+    function runInsertQuery(record {} 'object) returns sql:ExecutionResult|error {
+        sql:ParameterizedQuery query = sql:queryConcat(
+            `INSERT INTO `, self.tableName, ` (`,
+            self.getColumnNames(), ` ) `,
+            `VALUES `, self.getInsertQueryParams('object)
+        );
+        return check self.dbClient->execute(query);
+    }
+
+    // TODO: handle composite keys
+    function runReadByKeyQuery(anydata key) returns record {}|error {
+        sql:ParameterizedQuery query = sql:queryConcat(
+            `SELECT `, self.getColumnNames(), ` FROM `, self.tableName, ` WHERE `, check self.getGetKeyWhereClauses(key)
```

Comment:
This query is used to get all fields. Isn't it? 
Can't we use `*` instead of `self.getColumnNames()`?
```suggestion
            `SELECT * FROM `, self.tableName, ` WHERE `, check self.getGetKeyWhereClauses(key)
```

Reply:
Used the column names instead of `*` in case the table contains information not relevant to the entity directly (e.g. metadata fields - created_at, last_modified_at, soft delete flags etc.)